Flow.Launcher/Helper/WindowsInteropHelper.cs

```diff
@@ -8,156 +8,152 @@
 
 
-        const string WINDOW_CLASS_CONSOLE = "ConsoleWindowClass";
-        const string WINDOW_CLASS_WINTAB = "Flip3D";
-        const string WINDOW_CLASS_PROGMAN = "Progman";
-        const string WINDOW_CLASS_WORKERW = "WorkerW";
+    const string WINDOW_CLASS_CONSOLE = "ConsoleWindowClass";
+    const string WINDOW_CLASS_WINTAB = "Flip3D";
+    const string WINDOW_CLASS_PROGMAN = "Progman";
+    const string WINDOW_CLASS_WORKERW = "WorkerW";
 
-        public static bool IsWindowFullscreen()
+    public static bool IsWindowFullscreen()
+    {
+        //get current active window
+        IntPtr hWnd = GetForegroundWindow();
+
+        if (hWnd.Equals(IntPtr.Zero))
         {
-            //get current active window
-            IntPtr hWnd = GetForegroundWindow();
+            return false;
+        }
 
-            if (!hWnd.Equals(IntPtr.Zero))
-            {
-                //if current active window is NOT desktop or shell
-                if (!(hWnd.Equals(HWND_DESKTOP) || hWnd.Equals(HWND_SHELL)))
-                {
-                    StringBuilder sb = new StringBuilder(256);
-                    GetClassName(hWnd, sb, sb.Capacity);
-                    string windowClass = sb.ToString();
-
-                    //for Win+Tab (Flip3D)
-                    if (windowClass == WINDOW_CLASS_WINTAB)
-                    {
-                        return false;
-                    }
-
-                    RECT appBounds;
-                    GetWindowRect(hWnd, out appBounds);
-
-                    //for console (ConsoleWindowClass), we have to check for negative dimensions
-                    if (windowClass == WINDOW_CLASS_CONSOLE)
-                    {
-                        return appBounds.Top < 0 && appBounds.Bottom < 0;
-                    }
-
-                    //for desktop (Progman or WorkerW, depends on the system), we have to check 
-                    if (windowClass == WINDOW_CLASS_PROGMAN || windowClass == WINDOW_CLASS_WORKERW)
-                    {
-                        IntPtr hWndDesktop = FindWindowEx(hWnd, IntPtr.Zero, "SHELLDLL_DefView", null);
-                        hWndDesktop = FindWindowEx(hWndDesktop, IntPtr.Zero, "SysListView32", "FolderView");
-                        if (!hWndDesktop.Equals(IntPtr.Zero))
-                        {
-                            return false;
-                        }
-                    }
-
-                    Rectangle screenBounds = Screen.FromHandle(hWnd).Bounds;
-                    if ((appBounds.Bottom - appBounds.Top) == screenBounds.Height && (appBounds.Right - appBounds.Left) == screenBounds.Width)
-                    {
-                        return true;
-                    }
-                }
-            }
+        //if current active window is desktop or shell, exit early
+        if (hWnd.Equals(HWND_DESKTOP) || hWnd.Equals(HWND_SHELL))
+        {
+            return false;
+        }
```

Comment:
Reduced the two levels of nesting by returning early. I'm kind of on the fence with this one, too. I do think that generally reducing nesting improves readability, but what do you think for this specific case?

Reply:
Do you change it with rider refactor? I am a bit skeptical about whether it follows the same logic (my mind is not so clear atm).

Reply:
Yes, besides changing a comment